models/demos/t3000/falcon40b/tt/ops/fused_layernorm.py

```diff
@@ -0,0 +1,301 @@
+                    tt_dtype=self.model_config["LN_MLP_BIAS_DTYPE"],
+                )
+                self.ln_mlp_beta = ln_mlp_beta_host.to(device, self.model_config["LN_MLP_BIAS_MEMCFG"])
+                ttnn.experimental.tensor.dump_tensor(
+                    str(ln_mlp_bias_path),
+                    ln_mlp_beta_host,
+                )
+
+        self.layernorm_eps = config.layer_norm_epsilon
+
+        shard_spec_cores_grid = ttnn.experimental.tensor.CoreRangeSet(
+            {
+                ttnn.experimental.tensor.CoreRange(
+                    ttnn.experimental.tensor.CoreCoord(0, 0),
+                    ttnn.experimental.tensor.CoreCoord(7, 7),
+                ),
+            }
+        )
+        H = config.hidden_size  # 8192
+        self.sharded_memconfig = ttnn.experimental.tensor.MemoryConfig(
+            ttnn.experimental.tensor.TensorMemoryLayout.BLOCK_SHARDED,
+            ttnn.experimental.tensor.BufferType.L1,
+            ttnn.experimental.tensor.ShardSpec(
+                shard_spec_cores_grid,
+                [self.model_config["SEQ_LEN"] // 8, H // 8],  # 1024
+                ttnn.experimental.tensor.ShardOrientation.ROW_MAJOR,
+                False,
+            ),
+        )
+        self.width_sharded_memconfig = ttnn.experimental.tensor.MemoryConfig(
+            ttnn.experimental.tensor.TensorMemoryLayout.WIDTH_SHARDED,
+            ttnn.experimental.tensor.BufferType.L1,
+            ttnn.experimental.tensor.ShardSpec(
+                shard_spec_cores_grid,
+                [
+                    self.model_config["SEQ_LEN"],
+                    H // 64,
+                ],
+                ttnn.experimental.tensor.ShardOrientation.ROW_MAJOR,
+                False,
+            ),
+        )
+
+        self.prg_config = ttnn.experimental.operations.primary.LayerNormShardedMultiCoreProgramConfig(
+            compute_with_storage_grid_size=[8, 8],
+            subblock_w=min(8, H // 8),
+            block_h=self.model_config["SEQ_LEN"] // 32 // 8,
+            block_w=H // 32 // 8,
+            inplace=False,
+        )
+
+        self.interleaved_memconfig = ttnn.experimental.tensor.MemoryConfig(
+            ttnn.experimental.tensor.TensorMemoryLayout.INTERLEAVED,
+            ttnn.experimental.tensor.BufferType.L1,
+        )
+
+    def __call__(self, x: ttnn.experimental.tensor.Tensor) -> ttnn.experimental.tensor.Tensor:
+        # # OG layernorm
+        # out1 = ttnn.experimental.operations.primary.layernorm(
+        #     x, eps=self.layernorm_eps, gamma=self.ln_attn_gamma, beta=self.ln_attn_beta, output_mem_config=self.sharded_memconfig, program_config=self.prg_config
+        # )
+        # out2 = ttnn.experimental.operations.primary.layernorm(
+        #     x, eps=self.layernorm_eps, gamma=self.ln_mlp_gamma, beta=self.ln_mlp_beta, output_mem_config=self.sharded_memconfig, program_config=self.prg_config
+        # )
```

Comment:
Are these comments useful for debugging/experimenting in future, or just a leftover?

Reply:
I left those in, because the whole ops/fused_layernorm is for debugging only and I thought we might need this again (I still have a todo regarding bfp8 pcc issues on my list)